### ISSUE TITLE
How to check if a shell is running inside mintty

### ISSUE THREAD

**Issue by iamFIREcracker** [mintty/mintty#867]
Is there a way to check if a shell is running inside mintty?

Probably changing TERM to something different from `xterm` is not [ideal](https://github.com/mintty/mintty/issues/866), but what about setting a new environment variable instead (eg. MINTTY)?

With this, users could for example add the following to their .vimrc and conditionally change cursor shape based on the containing terminal:

```
if exists('$MINTTY')
    let &t_ti.="\e[1 q"
    let &t_SI.="\e[5 q"
    let &t_EI.="\e[1 q"
    let &t_te.="\e[0 q"
endif
```

**Comment by avih** [comment]
No, but you can configure mintty to use some other TERM value, and in your `~/.bashrc` check if it's mintty, and if yes replace it to the xterm value and export some other variable which you can later check in vim.

e.g. configure mintty to use TERM=MINTTY, then in your bashrc add:
```bash
if [ "$TERM" = "MINTTY" ]; then
    export TERM=xterm-256color
    export INSIDE_MINTTY=1
fi
```

**Comment by mintty** [comment]
> setting a new environment variable

This has also been discussed already, see #776. In that issue you'll also see how the terminal type can be detected. I will summarize a complete guide and add it to the wiki.

**Comment by mintty** [comment]
Released 3.0.1.
